client/my-sites/themes/theme-upload/index.jsx

```diff
@@ -271,5 +276,5 @@ export default connect(
 			installing: isInstallInProgress( state, siteId ),
 		};
 	},
-	{ uploadTheme, clearThemeUpload },
+	{ uploadTheme, clearThemeUpload, initiateThemeTransfer },
```

Comment:
> This way component would be Jetpack/WP.com agnostic. That is just a suggestion.

Explored this a little. Decided it will be best to do it at the same time as the transfer actions start using [dedicated state](https://github.com/Automattic/automated-transfer-api-contracts/blob/master/calypsoState.json) and selectors.